include/revng/BasicAnalyses/AdvancedValueInfo.h

```diff
@@ -627,18 +627,29 @@ class Expression {
     if (not DT.dominates(StopAt, LimitedStartBB))
       LimitedStartBB = StopAt;
 
-    SmallPtrSet<BasicBlock *, 4> IgnoreList = { StopAt };
+    // We want to stop our backward visit at the `StopAt` node, so we actually
+    // need to blacklist the predecessors of `StopAt`
+    SmallPtrSet<BasicBlock *, 4> IgnoreList;
+    for (BasicBlock *Pred : predecessors(StopAt)) {
+      IgnoreList.insert(Pred);
+    }
```

Comment:
Consider using `llvm::copy(predecessors(StopAt), std::inserter(IgnoreList.begin(), IgnoreList)`

Reply:
Now in 1d384e05daf8ce38e943a63a3267d8a72118b267 we make use of `llvm::copy` and `llvm::copy_if` a couple of lines below.